pkgs/development/tools/build-managers/meson/setup-hook.sh

```diff
@@ -31,7 +31,69 @@ mesonConfigurePhase() {
     runHook postConfigure
 }
 
+mesonBuildPhase() {
+    runHook preBuild
+
+    localFlagsArray=(
+        -j$NIX_BUILD_CORES
+        $mesonBuildFlags "${mesonBuildFlags[@]}"
+    )
+
+    echoCmd 'build flags' "${flagsArray[@]}"
+    TERM=dumb meson compile "${flagsArray[@]}"
```

Comment:
Why the `TERM`? `setup` does not set this.

Reply:
It's for Ninja (which `setup` doesn't run). https://github.com/NixOS/nixpkgs/pull/179027

Reply:
Might be nice to add a comment, or at least mention it in the commit message then.

Reply:
It's only needed for ninja, which could just inherit the default ninja handling.

Why set it for test/install? Those don't generally build anything anyway (it's already built when you called ninja) and TERM=dumb doesn't seem to be targeted at meson's test harness.

Reply:
I wasn't sure if test/install still ended up calling Ninja.  I expected at least install to.  If they don't I can remove it.